libs/sdk-core/src/error.rs

```diff
@@ -0,0 +1,30 @@
+use thiserror::Error;
+
+pub type SdkResult<T, E = SdkError> = Result<T, E>;
+
+/// Type of error returned by the SDK
+#[derive(Debug, Error)]
+pub enum SdkError {
+    #[error("Failed to connect to required services")]
+    ConnectFailed,
+
+    /// Generic error, that doesn't fit any of the other types
+    #[error("Breez SDK error: {err}")]
+    Generic { err: String },
+
+    #[error("Failed to initialize the SDK")]
+    InitFailed,
```

Comment:
I agree with your suggestion to have String payload for every error so the user can get a proper context so let's add it to all other errors.

Reply:
Done